### DISCUSSION TITLE
Basic Commands:

### DISCUSSION THREAD

**Discussion by lahiruroot** [lahiruroot/WUSL#4] [General]
1. What does the command git status do?
2. How do you stage changes using Git? What is the command?
3. What is the purpose of the git commit command?

**Comment by prasajanravishka** [comment]
### **Git Commands Summary:**

1.    Displays the state of the working directory and staging area, showing modified, staged, and untracked files.

2.    Use `git add` to stage changes:  
   - Stage a specific file: `git add <file-name>`  
   - Stage all changes: `git add .`  

3.    Records staged changes into the repository's history. Example:  
   - Quick commit with a message: `git commit -m "Your message"`  
   - Open editor for message: `git commit`

**Comment by PrasadWc** [comment]
1. shows the current state of your working directory and staging area. It lists(Changes not yet staged for commit, Staged changes ready to be committed, Untracked files.)
2. use the command git add
3. saves the staged changes into the local repository with a descriptive message.

**Comment by sandini0130** [comment]
1. What does git status do?
It shows the current state of your project, like changed files and files ready to be committed.
2. How do you stage changes?
Use the git add command:
Stage a file: git add <file-name>
Stage all files: git add .
3. What is git commit for?
It saves your staged changes permanently in Git with a message describing what you changed.
Example: git commit -m "Message"

**Comment by Nuwanil** [comment]
1.Displays the current status of your repository, showing changes staged for commit, changes not staged yet, and untracked files.
2.Stage Changes in Git
git add <file>
3. Saves the staged changes to the repository’s history with a descriptive message. It marks a point in the project's version history.

**Comment by chathurika11** [comment]
1) it shows the current state of your working directory and staging area.
2) git add
stage all changes - git add .
stage a specific file - git add filename
3)it saves your changes into Git’s history with a message describing the changes.
Think of it like taking a snapshot of your work at that moment
git commit -m "Describe changed"

**Comment by chathurika11** [comment]
1) it shows the current state of your working directory and staging area.
2) git add
stage all changes - git add .
stage a specific file - git add filename
3)it saves your changes into Git’s history with a message describing the changes.
Think of it like taking a snapshot of your work at that moment
git commit -m "Describe changed"

**Comment by Dilshan2003** [comment]
git status:
Displays the current state of your working directory and staging area.
Shows information about the current branch, untracked files, changes to be committed, modified files, and deleted files.
Staging Changes:
Use the command git add <file_name> to stage specific files.
To stage all changes at once, use git add ..
git commit:
Saves staged changes to the local repository.
Creates a snapshot of your changes with a descriptive message using the syntax: git commit -m "Your commit message here".

**Comment by Ashini248** [comment]
1. What does the command git status do?
git status shows the current state of your working directory and staging area.
It tells you:
Files that have been modified but not yet staged.
Files in the staging area ready to be committed.
Untracked files (new files not being tracked by Git).
Example:
       git status

2. How do you stage changes using Git?
To stage changes, you use the git add command. This prepares files for the next commit.
Commands:
        Stage a specific file:
        (bash)
        git add <file-name>

Stage all changes in the repository:
            (bash)
            git add .


3. What is the purpose of the git commit command?
The git commit command saves your staged changes into the local repository as a snapshot.
Each commit has a unique ID and message to describe the changes.

Command:
(bash)
git commit -m "Your commit message"

**Comment by pinipa515** [comment]
1. **`git status`**: Displays the current state of your working directory and staging area. It shows:
   - Changes that are staged for commit.
   - Unstaged changes in tracked files.
   - Untracked files in the directory.

2. **How to stage changes**: Use the `git add` command.  
   Example:  
   - To stage a specific file: `git add <filename>`  
   - To stage all changes: `git add .`

3. **Purpose of `git commit`**: Saves the staged changes as a new version in the repository's history, with a descriptive message explaining what the changes are. It creates a snapshot of your code at that point.  
   Example: `git commit -m "Your commit message"`

**Comment by ReshaniDananjana** [comment]
1.The git status command shows the current state of the working directory and staging area, including untracked files, changes to be committed, and changes not yet staged.
2.git add <file-name>  
   git add .      
3.The git commit command saves your staged changes to the local repository as a new commit, creating a snapshot of your progress.

**Comment by Chathushka04** [comment]
(1). The git status command shows the current state of your working directory and staging area. It displays:

*Untracked files (new files not added to Git),
*Modified files (changed but not staged),
*Staged files (ready to be committed),
*Branch status (current branch and its relation to the remote).

(2).  To stage changes in Git, use git add <file-name> to stage specific files, or git add . to stage all changes. You can also use git add -p to selectively stage parts of a file. This prepares changes for the next commit.

(3).  The `git commit` command saves staged changes to the local repository, creating a snapshot with a unique identifier and a commit message describing the changes. It records changes in the version history.
